syntax = "proto3";

import "google/protobuf/timestamp.proto";

option go_package = "/generated";



message UserDataWhirlpool {
    string uid = 1;
    bytes session = 2;
    string ownerKey = 3;
}

// TODO: user data surface

message UserCertificate {
    bytes token = 1;
    int32 seaPort = 2;
    int32 controlPort = 3;
}



message UserToken {
    string uid = 1;
    bytes session = 2;
    bool privileged = 3;
    optional google.protobuf.Timestamp subscription = 4;
}
